wpengine/atlas-content-modeler
# refactor: reduce duplication in field forms

Removes repeated form logic.

Defines a new base `Form.jsx` that renders additional fields from `MediaFields.jsx`, `NumberFields.jsx`, and `TextFields.jsx` based on the active field type.

Assumes that all fields need a name and identifier, and that additional fields will appear in the form's left column for now.

The Boolean field type has no extra fields so the `BooleanForm.jsx` file was removed.

As the name and id fields are common to all forms and rendered by the same `Form.jsx`, it's now possible to switch field type without losing edits to the field name.

Based on the `refactor/icons` branch and targets that to make the diff view better and reduce merge conflicts.

### To test
- Check field-specific form inputs still appear.
- Check `numberType`, `uploadType`, and `textLength` still appear in saved field data in `wp option get wpe_content_model_post_types`, for example:
 
```
      1615902339564 =>
      array (
        'type' => 'text',
        'id' => '1615902339564',
        'position' => '0',
        'name' => 'Text',
        'slug' => 'text',
        'textLength' => 'short',
      ),
      1615902343291 =>
      array (
        'type' => 'number',
        'id' => '1615902343291',
        'position' => '10000',
        'name' => 'Number',
        'slug' => 'number',
        'numberType' => 'integer',
      ),
      1615902349253 =>
      array (
        'type' => 'boolean',
        'id' => '1615902349253',
        'position' => '20000',
        'name' => 'Boolean',
        'slug' => 'boolean',
      ),
      1615902355094 =>
      array (
        'type' => 'media',
        'id' => '1615902355094',
        'position' => '30000',
        'name' => 'Media',
        'slug' => 'media',
        'uploadType' => 'multi',
      ),
```
